Unable to create sink while subscription_position is not specified

### Terraform Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

- 0.1.2 and 0.1.3

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

- pulsar_sink

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
locals {
  cloud_storage_sinks = {
    billing = {
      bucket  = "pulsar-billing"
      pattern = "persistent://billing/default/*"
      region  = "us-west-2"
    }
  }
}

resource "pulsar_sink" "cloud_storage_sink" {
  for_each = local.cloud_storage_sinks

  name                 = "cloud-storage-${each.key}"
  tenant               = "public"
  namespace            = "default"
  archive              = "builtin://cloud-storage"
  parallelism          = 1
  topics_pattern       = each.value.pattern
  auto_ack             = true
  cpu                  = 1
  ram_mb               = 1024
  disk_mb              = 10240
  cleanup_subscription = true
  configs              = jsonencode({
    bucket                = each.value.bucket
    region                = each.value.region
    role                  = "arn:aws:iam::123456789:role/pulsar-storage-connector-role"
    roleSessionName       = "name"
    provider              = "s3v2"
    formatType            = "json"
    partitionerType       = "partition"
    timePartitionPattern  = "yyyy-MM-dd"
    timePartitionDuration = "1h"
    batchSize             = 15000
    batchSizeMs           = 15000
  })
}
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

```
Stack trace from the terraform-provider-pulsar_v0.1.2 plugin:

panic: Error reading level config: '' expected type 'string', got unconvertible type 'utils.MessageID', value: '-1:-1:-1:-1'

goroutine 76 [running]:
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).get(0xc00008e100, {0xc00043c280, 0x1, 0x1}, 0x0?)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/resource_data.go:553 +0x328
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).getChange(0xc00011ec80?, {0xcb701a?, 0x15?}, 0x27?, 0x0?)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/resource_data.go:524 +0x1b8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).diffChange(0xc0000bcc30?, {0xcb701a?, 0xcab40b?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/resource_data.go:501 +0x76
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.diffString(0xc0000bcd78?, {0xcb701a, 0x15}, 0xc000470640, 0xc0000bcdd8, {0xddfa40, 0xc00008e100}, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/schema.go:1521 +0xba
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.diff(0x6?, {0xddd598, 0xc0000fa5d0}, {0xcb701a, 0x15}, 0xc000470640, 0xc00008e000, {0xddfa40?, 0xc00008e100}, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/schema.go:1103 +0x16b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.Diff(0xc00044d590, {0xddd598, 0xc0000fa5d0}, 0xc000517380, 0xc0000fb1d0, 0x0, {0xc7f140, 0xc000358b90}, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/schema.go:657 +0x333
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).SimpleDiff(0xdddd78?, {0xddd598?, 0xc0000fa5d0?}, 0xc000517380, 0xbc60a0?, {0xc7f140?, 0xc000358b90?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/resource.go:890 +0x6c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).PlanResourceChange(0xc000195a10, {0xddd4f0?, 0xc00011e700?}, 0xc0003ee7d0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/grpc_provider.go:741 +0x98c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).PlanResourceChange(0xc0003aed20, {0xddd598?, 0xc000373dd0?}, 0xc000506e00)
	github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/tf5server/server.go:779 +0x515
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_PlanResourceChange_Handler({0xc7ba60?, 0xc0003aed20}, {0xddd598, 0xc000373dd0}, 0xc00018df20, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:367 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001b6700, {0xde05f8, 0xc000003380}, 0xc00033d8c0, 0xc00044d6b0, 0x12a0668, 0x0)
	google.golang.org/grpc@v1.45.0/server.go:1282 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc0001b6700, {0xde05f8, 0xc000003380}, 0xc00033d8c0, 0x0)
	google.golang.org/grpc@v1.45.0/server.go:1619 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.45.0/server.go:921 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.45.0/server.go:919 +0x28a

Error: The terraform-provider-pulsar_v0.1.2 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

```

### Expected Behavior

<!--- What should have happened? --->

### Actual Behavior

<!--- What actually happened? --->

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

Able to create sink when `subscription_position` is specified.

The `subscription_position` should either be `Earliest` or `Latest`, but it's default value is using `utils.Earliest` instead.

--- a/pulsar/resource_pulsar_sink.go
+++ b/pulsar/resource_pulsar_sink.go
@@ -78,7 +78,7 @@ func init() {
 	resourceSinkDescriptions[resourceSinkCleanupSubscriptionKey] =
 		"Whether the subscriptions the functions created/used should be deleted when the functions was deleted"
 	resourceSinkDescriptions[resourceSinkSubscriptionPositionKey] =
-		"Pulsar source subscription position if user wants to consume messages from the specified location (Latest, Earliest)"
+		"Pulsar source subscription position if user wants to consume messages from the specified location (Latest, Earliest). Default to Earliest."
 	resourceSinkDescriptions[resourceSinkCustomSerdeInputsKey] =
 		"The map of input topics to SerDe class names (as a JSON string)"
 	resourceSinkDescriptions[resourceSinkCustomSchemaInputsKey] =
@@ -173,7 +173,7 @@ func resourcePulsarSink() *schema.Resource {
 			resourceSinkSubscriptionPositionKey: {
 				Type:        schema.TypeString,
 				Optional:    true,
-				Default:     utils.Earliest,
+				Default:     "Earliest",
 				Description: resourceSinkDescriptions[resourceSinkSubscriptionPositionKey],
 			},
 			resourceSinkCustomSerdeInputsKey: {

--- a/pulsar/resource_pulsar_sink.go
+++ b/pulsar/resource_pulsar_sink.go
@@ -77,8 +77,8 @@ func init() {
 		"Pulsar source subscription name if user wants a specific subscription-name for input-topic consumer"
 	resourceSinkDescriptions[resourceSinkCleanupSubscriptionKey] =
 		"Whether the subscriptions the functions created/used should be deleted when the functions was deleted"
-	resourceSinkDescriptions[resourceSinkSubscriptionPositionKey] =
-		"Pulsar source subscription position if user wants to consume messages from the specified location (Latest, Earliest). Default to Earliest."
+	resourceSinkDescriptions[resourceSinkSubscriptionPositionKey] = "Pulsar source subscription position if user wants " +
+		"to consume messages from the specified location (Latest, Earliest). Default to Earliest."
 	resourceSinkDescriptions[resourceSinkCustomSerdeInputsKey] =
 		"The map of input topics to SerDe class names (as a JSON string)"
 	resourceSinkDescriptions[resourceSinkCustomSchemaInputsKey] =

